unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,217 @@
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            self.redis_client = redis_client_auto_register
+            self.redis_client.is_active = True
+            self.redis_client.unregister(is_blocking=True)
+
+    def test_on_register(self, redis_client_auto_register):
+        """assigning the correct arguments to blocking command response and
+        checking by calling on_register function with correct message"""
+        message = {"data": {"device_uuid": TRANSACTION_ID, "transaction_id": TRANSACTION_ID}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._blocking_command_responses = \
+            {"register": {"transaction_id": f"{TRANSACTION_ID}"}}
+        redis_client_auto_register._on_register(message)
+
+    def test_on_unregister(self, redis_client_auto_register):
+        """assigning the correct arguments to blocking command response and
+        checking by calling on_unregister function with correct message"""
+        message = {"data": {"device_uuid": TRANSACTION_ID, "transaction_id": TRANSACTION_ID,
+                            "response": "success"}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._blocking_command_responses \
+            = {"unregister": {"transaction_id": f"{TRANSACTION_ID}"}}
+        redis_client_auto_register._on_unregister(message)
+
+    def test_on_unregister_throws_exception(self, redis_client_auto_register):
+        """checks if response is not success should throw exception """
+        with pytest.raises(RedisAPIException,
+                           match=f"Failed to unregister from market {AREA_ID}."
+                                 "Deactivating connection."):
+            message = {"data": {"device_uuid": TRANSACTION_ID, "transaction_id": TRANSACTION_ID,
+                                "response": "unsuccessful"}}
+            message["data"] = json.dumps(message["data"])
+            redis_client_auto_register._blocking_command_responses \
+                = {"unregister": {"transaction_id": f"{TRANSACTION_ID}"}}
+            redis_client_auto_register._on_unregister(message)
+
+    def test_on_event_or_response(self, redis_client_auto_register):
+        """assigning the correct arguments to blocking command response and
+        checking by calling on_event_or_response function with correct message"""
+        message = {"data": {"device_uuid": TRANSACTION_ID,
+                            "transaction_id": TRANSACTION_ID}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._on_event_or_response(message)
```

Comment:
What are you really testing here? I think the only way to test `_on_event_or_response` is to check if the ThreadPoolExecutor called `on_event_or_response`, but IMO you can skip it, and thus skip this whole test here.

Reply:
Sure